java/runtime/src/main/java/io/ray/runtime/gcs/GlobalStateAccessor.java

```diff
@@ -0,0 +1,67 @@
+package io.ray.runtime.gcs;
+
+import com.google.common.base.Preconditions;
+import java.util.List;
+
+/**
+ * `GlobalStateAccessor` is used for accessing information from GCS.
+ *
+ **/
+public class GlobalStateAccessor {
+  // NOTE(lingxuan.zlx): Native pointer is singleton in gcs state accessor, which means it can not
+  // be changed in cluster when redis or other storage accessor is fixed.
+  private static Long globalStateAccessorNativePtr = 0L;
+
+  public GlobalStateAccessor(String redisAddress, String redisPassword) {
```

Comment:
I think we can also change this Java class to a singleton. So we won't accidentally call this constructor while the previous one is not disconnected yet.
We can put the Reds address and password in the `connect` method and initialize the native object there. And in `disconnect`, we do disconnect as well as destroy the native object.


  public void disconnect() {
    this.nativeDisconnect(globalStateAccessorNativePtr);
  }
```

Reply:
GlobalStateAccessor is only used in gcsclient of java worker. Maybe we can refactor it in new PR.

> I think we can also change this Java class to a singleton. So we won't accidentally call this constructor while the previous one is not disconnected yet.
> We can put the Reds address and password in the `connect` method and initialize the native object there. And in `disconnect`, we do disconnect as well as destroy the native object.
> 
> public void disconnect() {
> this.nativeDisconnect(globalStateAccessorNativePtr);
> }